src/Extents.jl

```diff
@@ -206,4 +231,22 @@ end
 
 _unwrap(::Val{X}) where {X} = X
 
+# Bounds
+
+function _bounds_comparisons(f, ext1, ext2, strict)
+    _maybe_check_keys_match(ext1, ext2, strict) || return false
+    keys = _shared_keys(ext1, ext2)
+    if length(keys) == 0
+        return false # Otherwise `all` returns `true` for empty tuples
+    else
+        dimintersections = map(keys) do k
+            f(ext1[_unwrap(k)], ext2[_unwrap(k)])
+        end
+        return all(dimintersections)
+    end
+end
+
+_bounds_intersect(b1::Tuple, b2::Tuple) = (b1[1] <= b2[2] && b1[2] >= b2[1])# || (b2[1] <= b1[2] && b2[2] >= b1[1])
```

Comment:
Still commented out code here to remove that should be included? Doesn't seem to be complete yet.

Might be good to have a comment like
# abab or baba (which also matches abba and baab, but not aabb or bbaa)
and/or even destructure the tuples to (mina, maxa), (minb, maxb) for readability.

Note that this is sometimes written as the inverse `(maxa < minb) || (mina > maxb)`

Reply:
I think this wasn't actually needed as we assume the tuple is in ascending order.